sql/core/src/main/scala/org/apache/spark/sql/execution/AliasAwareOutputs.scala

```diff
@@ -16,16 +16,18 @@
  */
 package org.apache.spark.sql.execution
 
-import org.apache.spark.sql.catalyst.expressions.{Alias, Attribute, AttributeReference, Expression, NamedExpression}
+import org.apache.spark.sql.catalyst.expressions.{Alias, Attribute, AttributeReference, Expression, NamedExpression, SortOrder}
 import org.apache.spark.sql.catalyst.plans.physical.{HashPartitioning, Partitioning}
 
 /**
- * A trait that handles aliases in the `outputExpressions` to produce `outputPartitioning`
- * that satisfies output distribution requirements.
+ * A trait that handles aliases in the `outputExpressions` and `orderingExpressions` to produce
+ * `outputPartitioning` and `outputOrdering` that satisfy distribution and ordering requirements.
  */
-trait AliasAwareOutputPartitioning extends UnaryExecNode {
+trait AliasAwareOutputs extends UnaryExecNode {
```

Comment:
How about the name, `AliasAwareOutputPartitioningAndOrdering`?

Reply:
Yes, good idea.